store/sqlstore/supplier_reactions.go

```diff
@@ -103,6 +104,39 @@ func (s *SqlSupplier) ReactionGetForPost(ctx context.Context, postId string, hin
 	return result
 }
 
+func (s *SqlSupplier) ReactionsBulkGetForPosts(ctx context.Context, postIds []string, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	var reactions []*model.Reaction
+	query := `SELECT
+				*
+			FROM
+				Reactions
+			WHERE
+				PostId in (%v)
+			ORDER BY
+				CreateAt`
+
+	query = fmt.Sprintf(query, getPostIdsClause(postIds))
+
+	if _, err := s.GetReplica().Select(&reactions, query); err != nil {
+		result.Err = model.NewAppError("SqlReactionStore.GetForPost", "store.sql_reaction.bulk_get_for_post_ids.app_error", nil, "", http.StatusInternalServerError)
+	} else {
+		result.Data = reactions
+	}
+
+	return result
+}
+
+func getPostIdsClause(postIds []string) string {
+	var postIdsQuoted []string
+
+	for _, postId := range postIds {
+		postIdsQuoted = append(postIdsQuoted, fmt.Sprintf(`'%v'`, postId))
```

Comment:
Since we're just concatenating strings here, I'm fairly certain this is vulnerable to SQL injection. Instead of using `getPostIdsClause`, you should use the existing `MapStringsToQueryParams` helper since it uses named parameters which are properly escaped. You can see how it's used [in the emoji store](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/emoji_store.go#L134)